runtime/onert/backend/open_cl/Arguments.h

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __ONERT_BACKEND_GPU_CL_ARGUMENTS_H__
+#define __ONERT_BACKEND_GPU_CL_ARGUMENTS_H__
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "ClDevice.h"
+#include "GpuObject.h"
+#include "OpenclWrapper.h"
+
+#include "AccessType.h"
+#include "Types.h"
+#include "Util.h"
+
+namespace onert
+{
+namespace backend
+{
+namespace gpu_cl
+{
+
+class ArgumentsBinder
+{
+public:
+  virtual bool SetInt(const std::string &name, int value) = 0;
+  virtual bool SetFloat(const std::string &name, float value) = 0;
+  virtual bool SetHalf(const std::string &name, half value) = 0;
+  virtual ~ArgumentsBinder() = default;
+};
+
+class Arguments : public ArgumentsBinder
+{
+public:
+  Arguments() = default;
+  void AddFloat(const std::string &name, float value = 0.0f);
+  void AddHalf(const std::string &name, half value = half(0.0f));
+  void AddInt(const std::string &name, int value = 0);
+  void AddObjectRef(const std::string &name, AccessType access_type,
+                    GPUObjectDescriptorPtr &&descriptor_ptr);
+  void AddObject(const std::string &name, GPUObjectDescriptorPtr &&descriptor_ptr);
+
+  bool SetInt(const std::string &name, int value) override;
+  bool SetFloat(const std::string &name, float value) override;
+  bool SetHalf(const std::string &name, half value) override;
+  bool SetObjectRef(const std::string &name, const GPUObject *object);
+
+  bool Bind(cl_kernel kernel, int offset = 0);
+
+  void RenameArgs(const std::string &postfix, std::string *code) const;
+  bool Merge(Arguments &&args, const std::string &postfix);
+
+  bool AllocateObjects(CLContext *context);
+  void ReleaseCPURepresentation();
+  bool TransformToCLCode(const DeviceInfo &device_info,
+                         const std::map<std::string, std::string> &linkables, std::string *code);
+
+  // Move only
+  Arguments(Arguments &&args);
+  Arguments &operator=(Arguments &&args);
+  Arguments(const Arguments &) = delete;
+  Arguments &operator=(const Arguments &) = delete;
+
+  ~Arguments() override = default;
+
+private:
+  // friend flatbuffers::Offset<data::Arguments> Encode(
```

Comment:
Also.

Reply:
I think.. It is not need this right now. So I commented it. If it is need, i will add.